packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -0,0 +1,611 @@
+   *
+   * @default - a new log group is created if slow index logging is enabled
+   */
+  readonly slowIndexLogGroup?: logs.LogGroup;
+
+  /**
+   * Specify if Elasticsearch application logging should be set up.
+   *
+   * @default - false
+   */
+  readonly appLogEnabed?: boolean;
+
+  /**
+   * Log Elasticsearch application logs to this log group.
+   *
+   * @default - a new log group is created if app logging is enabled
+   */
+  readonly appLogGroup?: logs.LogGroup;
+}
+
+/**
+ * Whether the domain should encrypt data at rest, and if so, the AWS Key
+ * Management Service (KMS) key to use. Can only be used to create a new domain,
+ * not update an existing one
+ */
+export interface EncryptionAtRestOptions {
+  /**
+   * Specify true to enable encryption at rest.
+   */
+  readonly enabled?: boolean;
+
+  /**
+   * Supply if using KMS key for encryption at rest.
+   */
+  readonly kmsKey?: kms.Key;
+}
+
+/**
+ * Properties for a AWS Elasticsearch Domain.
+ */
+export interface DomainProps {
+  /**
+   * Domain Access policies.
+   *
+   * @default - No access policies.
+   */
+  readonly accessPolicies?: PolicyStatement[];
+
+  /**
+   * Additional options to specify for the Amazon ES domain.
+   *
+   * @default - no advanced options are specified
+   */
+  readonly advancedOptions?: { [key: string]: (string) };
+
+  /**
+   * `AWS::Elasticsearch::Domain.CognitoOptions`
+   */
+  readonly cognitoOptions?: CfnDomain.CognitoOptionsProperty | cdk.IResolvable;
+
+  /**
+   * Enforces a particular physical domain name.
+   *
+   * @default <generated>
+   */
+  readonly domainName?: string;
+
+  /**
+   * The configurations of Amazon Elastic Block Store (Amazon EBS) volumes that
+   * are attached to data nodes in the Amazon ES domain.
+   *
+   * @default - No EBS volumes attached.
+   */
+  readonly ebsOptions?: EbsOptions;
+
+  /**
+   * The cluster configuration for the Amazon ES domain.
+   *
+   */
+  readonly clusterConfig: ClusterConfig;
+
+  /**
+   * The Elasticsearch Version
+   *
+   */
+  readonly elasticsearchVersion: string;
+
+  /**
+   * Encryption at rest options for the cluster.
+   *
+   * @default - No encryption at rest
+   */
+  readonly encryptionAtRestOptions?: EncryptionAtRestOptions;
+
+
+  /**
+   * Configuration log publishing configuration options.
+   *
+   * @default - No logs are published
+   */
+  readonly logPublishingOptions?: LoggingOptions;
+
+
+  /**
+   * Specify true to enable node to node encryption.
+   *
+   * @default - Node to node encryption is not enabled.
+   */
+  readonly nodeToNodeEncryptionEnabled?: boolean;
+
+  /**
+   * The hour in UTC during which the service takes an automated daily snapshot
+   * of the indices in the Amazon ES domain. Only applies for Elasticsearch
+   * versions below 5.3.
+   */
+  readonly automatedSnapshotStartHour?: number;
+
+  /**
+   * `AWS::Elasticsearch::Domain.VPCOptions`
+   */
+  readonly vpcOptions?: CfnDomain.VPCOptionsProperty | cdk.IResolvable;
```

Comment:
Dont expose `Cfn` classes. Try to see if you can accept an `IVpc` here.

Also, the docs state:

> Provide one subnet ID for each Availability Zone that your domain uses. For example, you must specify three subnet IDs for a three Availability Zone domain

I would be nice to have this validation done at synth time.

Reply:
I have introduced my own interface. IVpc has a lot more configuration than ES uses. I reference ec2.ISecurityGroups and ec2.ISubnet.